lib/smack/SmackRep.cpp

```diff
@@ -154,8 +210,8 @@ void SmackRep::collectRegions(llvm::Module &M) {
 
 const Stmt* SmackRep::alloca(llvm::AllocaInst& i) {  
   const Expr* size = 
-    Expr::fn("$mul",lit(storageSize(i.getAllocatedType())),lit(i.getArraySize()));
-                       
+    Expr::fn(opName("$mul", {TRUNC(getIntSize(i.getOperand(0)))}), lit(storageSize(i.getAllocatedType())),lit(i.getArraySize()));
```

Comment:
I'm more worried about the semantics than efficiency at the moment. At the moment I just want to flag this issue, and make a decision on it after further discussion with Zvonimir.